### ISSUE TITLE
MailMunch plugin popup not displaying

### ISSUE THREAD

**Issue by ctec25** [pcfreak30/rocket-footer-js#6]
How can I exclude the js for Mailmunch which is popup.js from getting concatenated and moved to the footer?

**Comment by pcfreak30** [comment]
Use the attribute `data-no-minify="1"` on all related script tags

**Comment by ctec25** [comment]
Please any ideas on how to add the attribute to the tag of the script. It script is powered by jquery js

**Comment by pcfreak30** [comment]
Is it a plugin thats adding the jquery code to load Mailmunch? If so, please link it.

**Comment by ctec25** [comment]
Yes it is. https://wordpress.org/plugins/mailmunch/

**Comment by pcfreak30** [comment]
Looks like they are echoing out the script tag here https://plugins.svn.wordpress.org/mailmunch/trunk/public/class-mailmunch-public.php

```echo('<script data-cfasync="false" src="//a.mailmunch.co/app/v1/site.js" id="mailmunch-script" data-plugin="'.MAILMUNCH_PREFIX.'" data-mailmunch-site-id="'.$siteID.'" async></script>');```

I have added a commit to the plugin to support rockets exclude js settings. You need to (for now), overwrite `lib/Rocket/Footer/JS.php` in the plugins folder with https://raw.githubusercontent.com/pcfreak30/rocket-footer-js/master/lib/Rocket/Footer/JS.php

Then add `a.mailmunch.co` as an exclusion.

**Comment by pcfreak30** [comment]
In the future, I may put in a module to support this, but as it needs testing to see if it can be optimized or just needs to be excluded, will be sometime in the future.

**Comment by ctec25** [comment]
Thanks, but It doesn't seem to make any difference after overwrite lib/Rocket/Footer/JS.php and adding a.mailmunch.co to  'Excluded JavaScript Files' 

please note that i made sure the wp rocket and cloudflare caches are cleared

**Comment by pcfreak30** [comment]
A temp hack then would be to add the attribute to the mailmunch plugin code. The commit I made has not been tested, but I copied much of the logic from rocket itself.

You could also try excluding `//a.mailmunch.co/app/v1/site.js`.

I could add in a filter into the plugin to be used but then you would need to add PHP to functions to override the behavior.

**Comment by ctec25** [comment]
Hi, I have also tried to exclude //a.mailmunch.co/app/v1/site.js and //a.mailmunch.co/app/v1/popover.js
which are the scripts that are related to the mailmunch plugins but still get the same result. When I reload the page adding ?nocache the popup displays and those script show up . but without ?nocache at the end of the url, the two scripts are no where to be found meaning they have been combined and the exclusion does not work.

**Comment by pcfreak30** [comment]
Have you tried editing the plugin code, based on what I have given, to see if that actually fixes it?

**Comment by ctec25** [comment]
Yes, that was the first thing I did before excluding the scripts.

**Comment by pcfreak30** [comment]
Stupid question, but with a query string, have you verified the attribute was added? Have you purged the whole cache? The plugin stores transients of files. I would recommend my dedicated transient plugin to better control them.
